### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by clin2on3mun**
In this milestone:

-  i added species column field to the table.
- Added data to the table in the data.sql file.
- did transaction queries:
  set species to unspecified
updateSpecies
Rollback update
RevertsUpdates
Update species to pokemon and digimon where digimon is when name end with mon and pokemon to other animals
setSpeciesField
delete animals table 
DeleteAnimalsTable
Rollback to display the table with rows again
AfterRollback
Delete all animals born after Jan 1st, 2022.
Deleteanimalswith2022
Update all animals' weight to be their weight multiplied by -1.
updateWeight
Rollback to the savepoint
rollbacktoweight
Update all animals' weights that are negative to be their weight multiplied by -1 and commit transaction
Updateweightoremovenegative

did Group by and aggregate queries

How many animals are there?
countAnimals
How many animals have never tried to escape?
AnimalThatNeverEscaped
What is the average weight of animals?

AverageWeightofanimals

Who escapes the most, neutered or not neutered animals?
neuteredornot

What is the minimum and maximum weight of each type of animal?
maxmin

What is the average number of escape attempts per animal type of those born between 1990 and 2000?
averageEscapeattempts

### CODE REVIEW

**Review by Ibizugbe — COMMENTED** [review]
Hi @clin2on3mun , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a professional readme file, kindly take away every template guideline statement from your readme file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @clin2on3mun, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a `Live Demo` section in the readme without a live link. Kindly either deploy your app and provide a live link or remove the live demo section for a professional readme file


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: queries.sql [hunk 1]
```diff
@@ -8,5 +8,47 @@ SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 SELECT * FROM animals WHERE name NOT IN('Gabumon');
 
+/*transaction queries*/
+BEGIN; 
+UPDATE animals SET species='unspecified';
+ROLLBACK;
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to check if the changes were made before and after the rollback

File: queries.sql [hunk 2]
```diff
@@ -8,5 +8,47 @@ SELECT * FROM animals WHERE neutered=true;
...
 
+BEGIN;
+UPDATE animals
+SET species='digimon'
+WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon'
+WHERE species IS NULL;
+COMMIT;
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to check if the changes were made before and after the rollback

File: queries.sql [hunk 3]
```diff
@@ -8,5 +8,47 @@ SELECT * FROM animals WHERE neutered=true;
...
 
+BEGIN;
+vet_clinic=*# DELETE FROM ANIMALS;
+ROLLBACK;
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to check if the changes were made before and after the rollback

File: queries.sql [hunk 4]
```diff
@@ -8,5 +8,47 @@ SELECT * FROM animals WHERE neutered=true;
...
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth='jan-1-2022';
+
+SAVEPOINT update;
+UPDATE animals
+SET weight_kg= weight_kg*-1;
+ROLLBACK;
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to rollback to the savepoint `update`

File: queries.sql [hunk 5]
```diff
@@ -8,5 +8,47 @@ SELECT * FROM animals WHERE neutered=true;
...
+SELECT AVG(weight_kg) as Average_animal_weight FROM animals;
+SELECT neutered,MAX(escape_attempts) as most_escaped_attempts FROM animals
+GROUP BY neutered;
+SELECT species, AVG(weight_kg) AS average_weight FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN 'jan-1-1990' AND 'jan-1-2000'
+GROUP BY species;
```
Comment (alexander16108) [inline queries.sql:62]: please update the follow queries that are incorrect  👍🏾 
- SELECT COUNT(*) as animals FROM animals = 10
- SELECT AVG(weight_kg) as Average_animal_weight FROM animals; = 15.550000
- SELECT species, AVG(weight_kg) AS average_weight FROM animals
GROUP BY species;

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @clin2on3mun  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
screenshots added ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 6]
```diff
@@ -8,5 +8,52 @@ SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 SELECT * FROM animals WHERE name NOT IN('Gabumon');
 
+/*transaction queries*/
+BEGIN; 
+UPDATE animals SET species='unspecified';
+SELECT * as animals FROM animals;
+ROLLBACK;
```
Comment (addod19) [inline queries.sql:17]: - [ ] As mentioned by the previous reviewer, he said `check if the changes were made before and after the rollback`, here you have only checked for the `before`, how about the `after` the rollback?

File: queries.sql [hunk 7]
```diff
@@ -8,5 +8,52 @@ SELECT * FROM animals WHERE neutered=true;
...
+UPDATE animals SET species='pokemon'
+WHERE species IS NULL;
+SELECT * as animals FROM animals;
+COMMIT;
 
+BEGIN;
+DELETE FROM ANIMALS;
+SELECT * as animals FROM animals;
+ROLLBACK;
+
```
Comment (addod19) [inline queries.sql:32]: - [ ] Here also, you were asked to check the before and after the rollback, here you have only checked for the before

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @clin2on3mun, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Great job addressing some of the comments from the previous review :rocket:


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @clin2on3mun , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**